src/Microsoft.AspNet.Http/Features/QueryFeature.cs

```diff
@@ -41,11 +41,23 @@ void IFeatureCache.CheckFeaturesRevision()
         {
             if (_cachedFeaturesRevision != _features.Revision)
             {
-                _request = null;
-                _cachedFeaturesRevision = _features.Revision;
+                ResetFeatures();
             }
         }
 
+        public void UpdateFeatures(IFeatureCollection features)
```

Comment:
Added it to interface so they need to be implemented; could remove it from interface and just have them as methods? Though exposing the ability to invalidate the cache seems good; why you'd want to do it for these cases - I don't know...